crates/nargo/src/cli/compile_cmd.rs

```diff
@@ -59,7 +60,12 @@ pub fn compile_circuit<P: AsRef<Path>>(
     let mut driver = Resolver::resolve_root_config(program_dir.as_ref(), backend.np_language())?;
     add_std_lib(&mut driver);
 
-    driver.into_compiled_program(show_ssa, allow_warnings).map_err(|_| std::process::exit(1))
+    let config = CompileOptionsBuilder::default()
+        .allow_warnings(allow_warnings)
+        .show_ssa(show_ssa)
+        .build()
+        .unwrap();
```

Comment:
Perhaps we can separate this out int a struct within our clap Args from the beginning so that there is no need to create a separate struct to re-group these options later

Reply:
Sure thing, I think we'd have to add some form of builder later but agreed it's a bit overkill for now when all the flags are coming from a single place.

Will add a matching struct in nargo and we can `CompileOptions::from` it.